pandas/core/generic.py

```diff
@@ -3393,19 +3393,29 @@ def where(self, cond, other=np.nan, inplace=False, axis=None, level=None,
             else:
                 other = self._constructor(other, **self._construct_axes_dict())
 
+        if axis is None:
+            axis = 0
```

Comment:
I was trying to keep the original behavior. Your suggestion might cause an extra re-indexing if ``axis`` is set explictly to 0, but I can't find any situation where it gives a different result.